modin/engines/base/io/column_stores/feather_dispatcher.py

```diff
@@ -56,4 +56,6 @@ def _read(cls, path, columns=None, **kwargs):
             from pyarrow.feather import read_feather
 
             df = read_feather(path)
-        return cls.build_query_compiler(path, df.columns, use_threads=False)
+            # pyarrow.feather.read_feather doesn't support columns as pandas.Index
+            columns = list(df.columns)
+        return cls.build_query_compiler(path, columns, use_threads=False)
```

Comment:
For some reason, the `pyarrow.feather.read_feather` can correctly process the columns specified by `pandas.Index` only if it consists of one element. If columns are specified with `pandas.Index` that `len() > 1`, then an error occurs inside.

Since our implementation parallelizes reading by dividing columns between workers, we need to use more than 4 columns to observe this erroneous behavior (4 is the default number of ray workers)

Reply:
This seems like a bug in `feather`, should we also open an issue there?